ksqldb-functional-tests/src/test/resources/query-validation-tests/attr.json

```diff
@@ -0,0 +1,141 @@
+{
+  "comments": [
+    "Tests covering the use of ATTR aggregate function"
+  ],
+  "tests": [
+    {
+      "name": "valid attr with various scalar fields",
+      "statements": [
+        "CREATE STREAM input (k INT KEY, s STRING, i INT, bi BIGINT, d DOUBLE, bo BOOLEAN) WITH (kafka_topic='in', value_format='JSON');",
+        "CREATE TABLE out AS SELECT k, ATTR(s) AS s, ATTR(i) AS i, ATTR(bi) AS bi, ATTR(d) AS d, ATTR(bo) AS bo FROM input GROUP BY k;"
+      ],
+      "inputs": [
+        {"topic": "in", "key": 1, "value": {"s": "s", "i": 1, "bi": 1, "d": 1.0, "bo": true}},
+        {"topic": "in", "key": 1, "value": {"s": "s", "i": 1, "bi": 1, "d": 1.0, "bo": true}},
+        {"topic": "in", "key": 1, "value": {"s": "s", "i": 1, "bi": 1, "d": 1.0, "bo": true}}
+      ],
+      "outputs": [
+        {"topic": "OUT", "key": 1, "value": {"S": "s", "I": 1, "BI": 1, "D": 1.0, "BO": true}},
+        {"topic": "OUT", "key": 1, "value": {"S": "s", "I": 1, "BI": 1, "D": 1.0, "BO": true}},
+        {"topic": "OUT", "key": 1, "value": {"S": "s", "I": 1, "BI": 1, "D": 1.0, "BO": true}}
+      ]
+    },
+    {
+      "name": "valid attr with various non-scalar fields",
+      "statements": [
+        "CREATE STREAM input (k INT KEY, li ARRAY<INT>, s STRUCT<A STRING, B INT>) WITH (kafka_topic='in', value_format='JSON');",
```

Comment:
Does Attr work with Maps?  

Just curious.  Not sure it requires a QTT.

Reply:
It should! I tested struct (I was going to test a map nested in a struct but looks like I forgot to do that) because generally that's the most complicated one